crates/astria-sequencer-utils/src/genesis_parser.rs

```diff
@@ -0,0 +1,141 @@
+use std::{
+    fs::File,
+    path::Path,
+};
+
+use eyre::{
+    Result,
+    WrapErr,
+};
+use serde_json::{
+    to_writer_pretty,
+    Value,
+};
+
+use crate::config::Config;
+
+pub struct GenesisParser;
+
+impl GenesisParser {
+    /// Propagates json data from one json file to another.
+    ///
+    /// # Errors
+    ///
+    /// An `eyre::Result` is returned if either source genesis files cannot be opened,
+    /// or if the destination genesis file cannot be saved.
+    pub fn propagate_data(data: Config) -> Result<()> {
+        println!("loading genesis json data for propigation:");
+        println!("\tsource genesis file: {}", data.source_genesis_file);
+        println!(
+            "\tdestination genesis file: {}",
+            data.destination_genesis_file
+        );
+        // load sequencer genesis data
+        let source_genesis_file_path = File::open(data.source_genesis_file)
+            .wrap_err("failed to open sequencer genesis file")?;
+        let source_genesis_data: Value = serde_json::from_reader(&source_genesis_file_path)
+            .wrap_err("failed deserializing sequencer genesis state from file")?;
+
+        // load cometbft genesis data
+        let destination_genesis_file_path = File::open(data.destination_genesis_file.as_str())
```

Comment:
You will have 2 file handles open at the same time to the same file. 

Better read in this string in one go so that you only have one handle (this assumes that the input json will never be too large, which I think is reasonable:
```suggestion
        let dst_contents = std::fs::read_to_string(&data.destination_genesis_file).wrap_err("failed reading cometbft genesis file")?;
```